src/legacy/core_plugins/vis_type_vislib/public/components/options/metrics_axes/category_axis_panel.tsx

```diff
@@ -89,7 +89,7 @@ function CategoryAxisPanel(props: CategoryAxisPanelProps) {
         setValue={setAxis}
       />
 
-      {axis.show && <LabelOptions axis={axis} axesName="categoryAxes" index={0} {...props} />}
+      {axis.show && <LabelOptions axesName="categoryAxes" index={0} {...props} />}
```

Comment:
Since you touched this, could you please also optimize the `LabelOptions` component ?
This component is only relied on several params, but accepts all of `VisOptionsProps<BasicVislibParams>`.
Could be decreased (and could be helpful for performance, since will not cause additional re-rendering):

```
<LabelOptions 
    axisLabels={axis.labels}
    axisFilterCheckboxName={`xAxisFilterLabelsCheckbox${axis.id}`}
    setAxisLabel={setAxisLabel} />
```

Reply:
OK, don't worry about this anymore!
I prepared a PR with refactoring based on your current pr:
https://github.com/elastic/kibana/pull/59135